src/Compilers/CSharp/Test/Symbol/Symbols/StaticAbstractMembersInInterfacesTests.cs

```diff
@@ -5446,5 +5447,3416 @@ interface I14
                 Diagnostic(ErrorCode.ERR_BadAbstractShiftOperatorSignature, op).WithLocation(61, 35)
                 );
         }
+
+        [Fact]
+        public void ConsumeAbstractStaticMethod_01()
+        {
+            var source1 =
+@"
+interface I1
+{
+    abstract static void M01();
+
+    static void M02()
+    {
+        M01();
+        M04();
+    }
+
+    void M03()
+    {
+        this.M01();
+        this.M04();
+    }
+
+    static void M04() {}
+
+    protected abstract static void M05();
+}
+
+class Test
+{
+    static void MT1(I1 x)
+    {
+        I1.M01();
+        x.M01();
+        I1.M04();
+        x.M04();
+    }
+
+    static void MT2<T>() where T : I1
```

Comment:
Consider adding a test where a type parameter is constrained indirectly, for example:

`static void M2<T, U>() where T : I1, U : T`

Specifically that when we call methods off of 'U' the 'constrained' prefix refers to U instead of T